### ISSUE TITLE
Write up use case for Triquetrum and ICE integration

### ISSUE THREAD

**Issue by jayjaybillings** [eclipse/triquetrum#28]
@erwindl0 and @cxbrooks requested a good use case describing how ICE and Triquetrum can work with each other.

**Comment by jayjaybillings** [comment]
Ideally users could create a workflow using the RCP editor for Triquetrum and then that workflow would show up in the list of Items that they can use in ICE. We could test this by creating a Triquetrum workflow that provides some sort of simple data analysis routine and hook that into ICE.

I'm working up some sketches now of the best way to hook this up to ICE. However, the short term hack is to obtain the ICE Core service from the OSGi and then when a new workflow is created for Triquetrum it can just be registered with the Core via an adapter ItemBuilder and Item.

**Comment by erwindl0** [comment]
With the timely arrival of the [January](https://projects.eclipse.org/proposals/january) project proposal, I'll try to integrate that to access&use the datasets from the ICE data files.

**Comment by erwindl0** [comment]
Jay, could you already send some example data files?
And if possible a bit of doc on the data structures in there, plus some pointers to ICE APIs, code, ... that loads the files, and/or manipulates the data, and writes updated/new data?

thanks
erwin

**Comment by jayjaybillings** [comment]
Yes. I'll try to get you something in a couple of days. I'm on another project right now.

In the meantime, you might check out the org.eclipse.ice.vibe bundle or the org.eclipse.ice.reflectivity bundle.

**Comment by cxbrooks** [comment]
I'd like to be able to use the output of this issue in the Triquetrum talk on Tuesday, 3/8.  I have a dry run scheduled for Wednesday, 3/2, so I would like to get input no later than Tuesday, 3/1.
My notes say:
* Working on Adapters, need to find use cases where a Triq wf could come in handy.
* Technically, this is really easy. We need an example. If we create something in ICE that gets tranformer to a task in Triquetrum that then gets transformed to a Workflow. This should do something real for ICE. Maybe need to call back in to IC 
Basically, I'd like to have a slide about what Triquetrum can do for ICE.

**Comment by jayjaybillings** [comment]
I'm happy to provide that slide and to try to get the code going. However, I'm slammed lately so please please please keep on me about getting this information for you.

**Comment by erwindl0** [comment]
Jay, do you still plan to define a more comprehensive use case in the near future?
Or do we close this for now and stick to the simple demo as done for EclipseCon?

**Comment by jayjaybillings** [comment]
Let's close this for now. We are trying to sort some things out with ICE funding at the moment and we can look at the more comprehensive problem after that. Some of those funding things might actually let Greg and I get involved.
